File: 2020/day01.py
with open("day01_input.txt") as f:
	E = [int(x) for x in f]

n = len(E)

done = False
i = 0
while (not done and i<n):
	j = 0
	while (not done and j<n):
		if ((E[i]+E[j])==2020):
			print("Part 1: {:d}".format(E[i]*E[j]))
			done = True
		j = j + 1
	i = i + 1

done = False
i = 0
while (not done and i<n):
	j = 0
	while (not done and j<n):
		k = 0
		while (not done and k<n):
			if ((E[i]+E[j]+E[k])==2020):
				print("Part 2: {:d}".format(E[i]*E[j]*E[k]))
				done = True
			k = k + 1
		j = j + 1
	i = i + 1

File: 2020/day01.c
#include <stdio.h>
#include <stdlib.h>
#include <string.h>

long int plg_filelen(char *filename) {
  FILE *fid = fopen(filename, "r");
  fseek(fid, 0, SEEK_END);
  long int len = ftell(fid);
  fclose(fid);
  return len;
}

char * plg_readfile(char *filename, long int fsize) {
  char *buf = malloc(fsize + 1);
  FILE *fid = fopen(filename, "r");
  fread(buf, fsize, 1, fid);
  fclose(fid);
  return buf;
}

int main(int argc, char *argv[]) {
	long int n = plg_filelen("day01_input.txt");
	// printf("file len is %ld\n", n);
	char *buf = malloc(sizeof(char)*n);
	buf = plg_readfile("day01_input.txt", n);

	for (int i=0; i<n; i++) {
		printf("%c", buf[i]);
	}

	free(buf);
	return 0;
}
